File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const check = new Array(priorities.length).fill(false);
    const result = new Map();
    let answer = 9;
    let i;

    while (answer) {
        if (result.size) {
            i = Array.from(result.keys()).pop();
        } else {
            i = 0;
        }

        for (; i < priorities.length; i++) {
            if (!check[i] && priorities[i] === answer) {
                result.set(i, priorities[i]);
                check[i] = true;
            }

            if (i === priorities.length - 1 && result.size) {
                const loop = Array.from(result.keys()).pop();

                for (let j = 0; j < loop; j++) {
                    if (!check[j] && priorities[j] === answer) {
                        result.set(j, priorities[j]);
                        check[j] = true;
                    }
                }
            }
        }

        answer -= 1;
    }

    return Array.from(result.keys()).indexOf(location) + 1;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    priority = max(priorities)
    point = priorities.index(priority)
    answer = []

    while priority > 0:
        if len(answer):
            point = answer[-1]

        for i in range(point, len(priorities)):
            if priorities[i] == priority:
                answer.append(i)

            if i == len(priorities) - 1:
                for j in range(0, point):
                    if priorities[j] == priority:
                        answer.append(j)

        priority -= 1

    return answer.index(location) + 1